### ISSUE TITLE
False positive with `use of moved value` error

### ISSUE THREAD

**Issue by qarmin** [intellij-rust/intellij-rust#10439]
## Environment

* **IntelliJ Rust plugin version:** 0.4.193.5352-231
* **Rust toolchain version:** nice version - 1.69
* **IDE name and version:** IntelliJ IDEA 2023.1.1 (Community Edition)
* **Operating system:** Ubuntu 23.04

## Problem description
Simple code which compiles properly with some warnings
```
    fn aa() -> (Vec<i32>, Vec<i32>) {
        let mut bb = vec![1];
        let mut cc = vec![1];
        if true {
            (bb, cc) = (vec![2], vec![3])
        }
        (bb, cc)
    }
```
shows error in IDE
```
Use of moved value 
```

![Screenshot from 2023-05-06 19-17-02](https://user-images.githubusercontent.com/41945903/236638090-8c191c5e-2ac3-4d9e-be21-8a7dcad8b756.png)

**Comment by qarmin** [comment]
also with 
```
    fn aa() -> (Vec<i32>, Vec<i32>) {
        let mut bb;
        let mut cc;
        if true {
            (bb, cc) = (vec![2], vec![3])
        } else {
            (bb, cc) = (vec![2], vec![3])
        }
        (bb, cc)
    }
```
I got error `Use of possibly uninitialized variable`

**Comment by vlad20012** [comment]
Duplicate #10427

**Comment by qarmin** [comment]
Is this really duplicate?

From your response https://github.com/intellij-rust/intellij-rust/issues/10427#issuecomment-1534673947
```
 The issue is 100% reproducible on the latest Rust nightly toolchain, but not on the stable toolchain.
```
So I think that this is a little different issue, because happens on stable toolchain
